src/client/Microsoft.Identity.Client/TelemetryCore/Http/HttpTelemetryManager.cs

```diff
@@ -103,13 +102,13 @@ public string GetCurrentRequestHeader(ApiEvent eventInProgress)
                 return string.Empty;
             }
 
-            eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.ApiIdConstStrKey, out string apiId);
-            eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.CacheInfoKey, out string cacheInfo);
-            eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.RegionUsed, out string regionUsed);
-            eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.RegionSource, out string regionSource);
-            eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.RegionOutcome, out string regionOutcome);
-            eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.IsTokenCacheSerializedKey, out string isTokenCacheSerialized);
-            eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.IsLegacyCacheEnabledKey, out string isLegacyCacheEnabled);
+            string apiId = eventInProgress.ApiIdString;
+            string cacheInfo = eventInProgress.CacheInfoString;
+            string regionUsed = eventInProgress.RegionUsed;
+            string regionSource = eventInProgress.RegionAutodetectionSourceString;
+            string regionOutcome = eventInProgress.RegionOutcomeString;
+            string isTokenCacheSerialized = eventInProgress.IsTokenCacheSerializedString;
+            string isLegacyCacheEnabled = eventInProgress.IsLegacyCacheEnabledString;
```

Comment:
wondering if there is a benefit to doing these string operations here? Perhaps that is not needed allocations?

Reply:
No benefit. I intended for this PR to have only code removal and as little other changes as possible, so reviewing is easier and nothing gets lost. I'm thinking of refactoring this in a separate PR and maybe even split this PR into smaller ones (although that might be difficult = lots of class dependencies).